pch/main-pch.hpp

```diff
@@ -56,20 +67,46 @@
 #include <random>
 #include <ratio>
 #include <regex>
+#include <scoped_allocator>
 #include <set>
 #include <sstream>
 #include <stack>
 #include <stdexcept>
+#include <streambuf>
 #include <string>
 #include <string_view>
 #include <system_error>
 #include <thread>
 #include <tuple>
 #include <type_traits>
+#include <typeindex>
 #include <typeinfo>
 #include <unordered_map>
 #include <unordered_set>
 #include <utility>
 #include <vector>
-
 #include <flatbuffers/flexbuffers.h>
+
+#if defined(_MSC_VER)
+#   include "platform_win.h"
+#endif
+
+#if defined(TILES)
+#   if defined(USE_VCPKG)
+#      include <SDL2/SDL.h>
+#      include <SDL2/SDL_image.h>
+#      include <SDL2/SDL_ttf.h>
+#      include <SDL2/SDL_version.h>
+#      ifdef SDL_SOUND
+#          include <SDL2/SDL_mixer.h>
+#      endif
+#   else
+#      include <SDL.h>
+#      include <SDL_image.h>
+#      include <SDL_ttf.h>
+#      include <SDL_version.h>
+#      ifdef SDL_SOUND
+#          include <SDL_mixer.h>
+#      endif
+#   endif
+#endif
```

Comment:
Maybe include `sdl_wrapper.h` instead? `SDL_MAIN_HANDLED` is already defined there, and it is better to keep the SDL include logic in a single place.